dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,293 @@
+            X_copy[:, -1] = y
+
+            # make the matrix that will hold the correlations
+            correlations = np.memmap(
+                fout,
+                dtype=np.result_type(X, y),
+                mode="w+",
+                shape=(X_copy.shape[1], X_copy.shape[1]),
+                )
+
+            # main calculation. It modifies X in-place
+            corrcoef(X_copy, out=correlations, rowvar=False, copy=False)
+
+            # we don't care about the direction of correlation in terms of
+            # the penalty/quality
+            np.absolute(correlations, out=correlations)
+
+            # our objective
+            np.fill_diagonal(correlations, -correlations[:, -1] * alpha)
+
+            # Note: the full symmetric matrix (with both upper- and lower-diagonal
+            # entries for each correlation coefficient) is retained for consistency with
+            # the original formulation from Milne et al.
+            it = np.nditer(correlations[:-1, :-1], flags=['multi_index'], op_flags=[['readonly']])
+            cqm.set_objective((*it.multi_index, x) for x in it)
+
+        return cqm
+
+    def fit(
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: typing.Optional[float] = None,
+        num_features: typing.Optional[int] = None,
+        time_limit: typing.Optional[float] = None,
+    ) -> SelectFromQuadraticModel:
+        """Select the features to keep.
+
+        Args:
+            X:
+                Feature vectors formatted as a numerical 2D array-like.
+            y:
+                Class labels formatted as a numerical 1D array-like.
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.  `alpha=1` places all weight on
+                relevance and selects all features, whereas `alpha=0` places all
+                weight on redundancy and selects no features.
+                Defaults to the value provided to the constructor.
+            num_features:
+                The number of features to select.
+                Defaults to the value provided to the constructor.
+            time_limit:
+                The time limit for the run on the hybrid solver.
+                Defaults to the value provided to the constructor.
+
+        Returns:
+            This instance of `SelectFromQuadraticModel`.
+        """
+        X = np.atleast_2d(np.asarray(X))
+        if X.ndim != 2:
+            raise ValueError("X must be a 2-dimensional array-like")
+
+        # y is checked by the correlation method function
+
+        if alpha is None:
+            alpha = self._alpha
+        # alpha is checked by the correlation method function
+
+        if num_features is None:
+            num_features = self._num_features
+        # num_features is checked by the correlation method function
+
+        # time_limit is checked by the LeapHybridCQMSampelr
```

Comment:
Do you think the error message given by the CQM solver is useful enough here for someone who has never run the CQM solver directly?

Reply:
I expect most users to not specify it at all, which will then use the default set by `LeapHybridCQMSampler`. For those who are setting it, I think they'll be able to parse the error.